contracts/service-registry/src/state.rs

```diff
@@ -195,6 +214,64 @@ pub fn remove_service_override(
     Ok(())
 }
 
+pub fn update_verifier_authorization_status(
+    storage: &mut dyn Storage,
+    service_name: &ServiceName,
+    auth_state: &AuthorizationState,
+    verifiers: &[Addr],
+) -> error_stack::Result<(), ContractError> {
+    let mut authorized_count_change = 0i32;
+
+    for verifier_addr in verifiers.iter() {
+        VERIFIERS
+            .update(
+                storage,
+                (service_name, verifier_addr),
+                |existing_verifier| -> std::result::Result<Verifier, ContractError> {
+                    let previous_state = existing_verifier
+                        .as_ref()
+                        .map(|verifier| verifier.authorization_state.clone());
+
+                    update_auth_verifier_count_change_tracker(
+                        &mut authorized_count_change,
+                        &previous_state,
+                        auth_state,
+                    )?;
+
+                    let new_verifier = match existing_verifier {
+                        Some(mut verifier) => {
+                            verifier.authorization_state = auth_state.clone();
```

Comment:
don't borrow a type just to clone it later, prefer taking ownership of the object (update_verifier_authorization_status should take a reference to auth_state)